cmd/brew-services.rb

```diff
@@ -369,8 +383,11 @@ def started_as
 
   # Get current PID of daemon process from launchctl.
   def pid
-    status = `#{ServicesCli.launchctl} list | grep #{label} 2>/dev/null`.chomp
-    return $1.to_i if status =~ /\A([\d]+)\s+.+#{label}\z/
+    return $1.to_i if status =~ status_regexp
+  end
+
+  def exit_code
+    return $2.to_i if status =~ status_regexp
```

Comment:
An edge-case I noticed with local testing: if I `brew services list` then any services running as `root` will show up as `error` because they aren't listed in the non-root `brew services list` output. Not sure how best to handle that, though.